src/speech_node/speech.py

```diff
@@ -26,54 +26,133 @@ class bcolors:
 
 
 class TTS(object):
-    def __init__(self):
+    """
+    Bridge from TTS service calls and topic messages to Toyota TTS, introducing a buffer to handle all requests in
+    receiving order
+    """
+    def __init__(self, rate):
+        """
+        Constructor
+        :param rate: ROS parameter for spin-rate
+        """
+
         self.samples_path = rospy.get_param("~samples_path", "~/MEGA/media/audio/soundboard")
+        self.rate = rate
+
+        # buffer and goal state
```

Comment:
Suggestion: make instance attributes 'protected' by starting with an underscore (although not that important here, it is useful in larger projects)